riscv/processor.cc

```diff
@@ -1523,20 +1523,20 @@ reg_t processor_t::get_csr(int which, insn_t insn, bool write, bool peek)
       ret((VU.vxsat << VCSR_VXSAT_SHIFT) | (VU.vxrm << VCSR_VXRM_SHIFT));
     case CSR_INSTRET:
     case CSR_CYCLE:
+      if (state.v && !ctr_v_ok)
+        goto throw_virtual;
       if (!ctr_ok)
         goto throw_illegal;
-      if (!ctr_v_ok)
-        goto throw_virtual;
       ret(state.minstret);
     case CSR_MINSTRET:
     case CSR_MCYCLE:
       ret(state.minstret);
     case CSR_INSTRETH:
     case CSR_CYCLEH:
+      if (state.v && !ctr_v_ok)
```

Comment:
What about the check of `xlen`? Shouldn't that be higher priority than the check of `ctr_v_ok`? (It is for `hpmcounterh`)

Reply:
Agreed, since the CSRs shouldn't exist for RV64, hence should raise illegal-instruction traps.

Reply:
Ahh, good catch. I missed this case. I will update this PR.